tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py

```diff
@@ -56,194 +56,144 @@
 
-from wazuh_testing.tools.configuration import load_wazuh_configurations
-import wazuh_testing.generic_callbacks as gc
-import wazuh_testing.logcollector as logcollector
-from wazuh_testing.tools.monitoring import WINDOWS_AGENT_DETECTOR_PREFIX, FileMonitor, LOG_COLLECTOR_DETECTOR_PREFIX
-from wazuh_testing.tools import get_service, LOG_FILE_PATH
+from wazuh_testing.tools.configuration import load_configuration_template, get_test_cases_data
 from tempfile import gettempdir
-from wazuh_testing.tools.utils import lower_case_key_dictionary_array
 from wazuh_testing.tools.services import control_service
+from wazuh_testing.modules.logcollector import LOG_COLLECTOR_PREFIX, WINDOWS_AGENT_PREFIX, \
+                                               GENERIC_CALLBACK_ERROR_ANALYZING_FILE, \
+                                               GENERIC_CALLBACK_ERROR_ANALYZING_EVENTCHANNEL, \
+                                               GENERIC_CALLBACK_ERROR_ANALYZING_MACOS
+from wazuh_testing.modules.logcollector import event_monitor as evm
+from wazuh_testing.processes import check_if_daemons_are_running
 
-LOGCOLLECTOR_DAEMON = "wazuh-logcollector"
-prefix = LOG_COLLECTOR_DETECTOR_PREFIX
 
 # Marks
 pytestmark = pytest.mark.tier(level=0)
 
-# Configuration
-no_restart_windows_after_configuration_set = True
-test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
-configurations_path = os.path.join(test_data_path, 'wazuh_basic_configuration.yaml')
-
-wazuh_component = get_service()
-first_macos_log_process = False
+prefix = LOG_COLLECTOR_PREFIX
 macos_process_timeout_init = 10
+daemon = 'wazuh-logcollector'
 
-wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+# Reference paths
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, 'configuration_template')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
 
+# ------------------------------------------------ TEST_ACCEPTED_VALUES ------------------------------------------------
+# Configuration and cases data
+t1_configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_only_future_events.yaml')
+t1_cases_path = os.path.join(TEST_CASES_PATH, 'cases_only_future_events.yaml')
 temp_file_path = os.path.join(gettempdir(), 'testing.log')
 
+# Accepted values test configurations (t1)
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(t1_cases_path)
 
-log_format_list = ['syslog', 'json', 'snort-full', 'mysql_log', 'postgresql_log', 'nmapg', 'iis', 'djb-multilog',
-                   'multi-line:3', 'squid', 'audit']
-tcases = []
+if sys.platform == 'win32':
+    prefix = WINDOWS_AGENT_PREFIX
+    daemon = 'wazuh-agent.exe'
 
 
-if sys.platform == 'win32':
-    prefix = WINDOWS_AGENT_DETECTOR_PREFIX
-    log_format_list += ['eventchannel']
-elif sys.platform == 'darwin':
-    log_format_list += ['macos']
+def remove_item_from_list(item):
```

Comment:
Same as before. If each os has its own cases mybe is better to create a test_case file for each one.

Reply:
Done in [here](https://github.com/wazuh/wazuh-qa/pull/3486/commits/5ad95d105667db211f44fcdd35ffcb640efe9d08)